cmd/dogstatsd/subcommands/start/command.go

```diff
@@ -191,27 +222,9 @@ func RunAgent(ctx context.Context, cliParams *CLIParams, config config.Component
 		log.Debugf("Health check listening on port %d", healthPort)
 	}
 
-	opts := aggregator.DefaultAgentDemultiplexerOptions()
-	opts.UseOrchestratorForwarder = false
-	opts.UseEventPlatformForwarder = false
-	opts.EnableNoAggregationPipeline = config.GetBool("dogstatsd_no_aggregation_pipeline")
-	hname, err := hostname.Get(context.TODO())
-	if err != nil {
-		log.Warnf("Error getting hostname: %s", err)
-		hname = ""
-	}
-	log.Debugf("Using hostname: %s", hname)
-	demux := aggregator.InitAndStartAgentDemultiplexer(log, forwarder, opts, hname)
-	demux.AddAgentStartupTelemetry(version.AgentVersion)
-
-	// setup the metadata collector
-	components.MetaScheduler = metadata.NewScheduler(demux) //nolint:staticcheck
-	if err = metadata.SetupMetadataCollection(components.MetaScheduler, []string{"host"}); err != nil {
```

Comment:
This call seems to be absent from the new code, is it expected?

Reply:
Yes, the `host` metadata is now a component and no longer require all the init and setup phase.